floats.go

```diff
@@ -69,9 +69,8 @@ func AddScaledTo(dst, y []float64, alpha float64, s []float64) []float64 {
 	if len(dst) != len(s) || len(dst) != len(y) {
 		panic("floats: lengths of slices do not match")
 	}
-	for i, val := range s {
-		dst[i] = y[i] + alpha*val
-	}
+	copy(dst, y)
```

Comment:
Is it too much to have both? The current version shows up constantly in
blas,  whereas the three element argument just shows up here (and lots of
code that calls this,  especially in optimize).
On Jan 11, 2015 11:31 AM, "Fazlul Shahriar" <notifications@github.com>
wrote:

> In floats.go
> <https://github.com/gonum/floats/pull/30#discussion-diff-22769234>:
>
> > @@ -69,9 +69,8 @@ func AddScaledTo(dst, y []float64, alpha float64, s []float64) []float64 {
> >  	if len(dst) != len(s) || len(dst) != len(y) {
> >  		panic("floats: lengths of slices do not match")
> >  	}
> > -	for i, val := range s {
> > -		dst[i] = y[i] + alpha*val
> > -	}
> > +	copy(dst, y)
>
> There's no reason to have a copy here. I can change DaxpyUnitary to take a
> destination argument: DaxpyUnitary(alpha, x, y, z) where z[i] =
> alpha*x[i] + y[i]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gonum/floats/pull/30/files#r22769234>.
>